From 9d98db67fcfdc508cf0cb7ddb02c9db0d5e077a4 Mon Sep 17 00:00:00 2001
From: Mikko Valkonen <msvalkon@gmail.com>
Date: Tue, 30 Sep 2014 14:21:25 +0300
Subject: [PATCH] Fix for issues #488 and #483

Previously the builds were failing because the postprocess_cython_output
function generated the following comment:

/* Generated by Cython 0.21 */ */

The fix makes sure that an ending comment block will only be added if
the line does not already end with one after the split.
---
 util/cythonpp.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/util/cythonpp.py b/util/cythonpp.py
index a82f9fc..9dfd8fd 100755
--- a/util/cythonpp.py
+++ b/util/cythonpp.py
@@ -553,13 +553,15 @@ def postprocess_cython_output(filename, banner):
     # 3) remove timestamp in cython's header so that different timestamps do not
     #    confuse merger
     result = ['/* %s */\n' % (banner)]
-
     input = open(filename)
     firstline = input.readline()
 
     if firstline.strip().lower().startswith('/* generated by cython ') and firstline.strip().endswith('*/'):
         line = firstline.strip().split(' on ', 1)[0]
-        result.append(line + ' */')
+        # Issue #488 - This check must be done, or we might end up in situation
+        # where line already ends with a */ and causes a compilation error.
+        if not line.endswith(' */'):
+            result.append(line + ' */')
     else:
         result.append(firstline)
 
